morebits.js

```diff
@@ -3563,6 +3563,41 @@ Morebits.wikitext.page.prototype = {
 	}
 };
 
+/**
+ * *********** Morebits.userspaceLogger ************
+ * Handles logging actions to a userspace log, used in
+ * twinklespeedy and twinkleprod.
+ */
+
+Morebits.userspaceLogger = function(logPageName) {
+	this.initialText = '';
+	this.headerLevel = 3;
+
+	this.log = function(logText, summaryText) {
```

Comment:
Given that this only has one method, I would suggest instead having something like
```
Morebits.userspaceLogger = function(logPageName, logText, summaryText) {
	...
}
```
that handles everything instead of an object with only one needed function

Reply:
It isn't just these 3, there's also the public props `initialText` and `headerLevel` (though the latter is used always with the default value in TW). A function with 4-5 parameters is pretty bad design, akin to the pathetic `Morebits.wiki.api` constructor. But that's just my take.

Reply:
No rush @siddharthvp but I am curious what you think of this approach.